### PR TITLE
Security docs update

### PR DESCRIPTION
**Pull request by octocrock**
This update adds a section to Hardening Octopus for containers and removes the "containers don't do https" sentence on the container docs.

### CODE REVIEW

File: docs/administration/security/hardening-octopus.md [hunk 1]
```diff
@@ -253,6 +253,23 @@ The TCP ports listed below are defaults, and can be changed if required - refer
 |WinRM-HTTP|`TCP 5985`|Octopus Server|Anywhere|DENY|Prevents attackers from using the Octopus Server as a beachhead into your network via unsecured WinRM.|
 |All outbound|`ALL`|Octopus Server|Anywhere|???|Depends on how fine-grained control you want over what your Octopus Server can do. It also depends on where your workers are, and where you are deploying to. Allowing all outbound traffic is a good place to start, then perform network analysis to decide on your next step.|
 
+## Harden your containers
+
+If you run an [Octopus Deploy container](/docs/installation/octopus-in-container/index.md) these are a few of the steps you can take to secure it.
```
Comment (wordlee) [inline docs/administration/security/hardening-octopus.md]: I'd make the language a bit more certain here:
```suggestion
If you run an [Octopus Deploy container](/docs/installation/octopus-in-container/index.md), you need to take the following steps to secure it: 
```
If we're worried about giving users the impression these are the only steps they need to take, we can `in addition to your usual security measure for running apps out of containers, you need to...

File: docs/administration/security/hardening-octopus.md [hunk 2]
```diff
@@ -253,6 +253,23 @@ The TCP ports listed below are defaults, and can be changed if required - refer
 |WinRM-HTTP|`TCP 5985`|Octopus Server|Anywhere|DENY|Prevents attackers from using the Octopus Server as a beachhead into your network via unsecured WinRM.|
 |All outbound|`ALL`|Octopus Server|Anywhere|???|Depends on how fine-grained control you want over what your Octopus Server can do. It also depends on where your workers are, and where you are deploying to. Allowing all outbound traffic is a good place to start, then perform network analysis to decide on your next step.|
 
+## Harden your containers
+
+If you run an [Octopus Deploy container](/docs/installation/octopus-in-container/index.md) these are a few of the steps you can take to secure it. 
+
+- Store your containers on a separate partition.
+- Assign resources carefully
```
Comment (wordlee) [inline docs/administration/security/hardening-octopus.md]:
```suggestion
- Assign resources carefully:
```

File: docs/administration/security/hardening-octopus.md [hunk 3]
```diff
@@ -253,6 +253,22 @@ The TCP ports listed below are defaults, and can be changed if required - refer
...
 |All outbound|`ALL`|Octopus Server|Anywhere|???|Depends on how fine-grained control you want over what your Octopus Server can do. It also depends on where your workers are, and where you are deploying to. Allowing all outbound traffic is a good place to start, then perform network analysis to decide on your next step.|
 
+## Harden your containers
+
+If you run an [Octopus Deploy container](/docs/installation/octopus-in-container/index.md) these are a few of the steps you can take to secure it. 
+
+- Store your containers on a separate partition.
+- Assign resources carefully:
+  - Consider pinning CPUs to namespaces in order to give them a boundary. 
+  - Consider the amount of memory required, too much and the container is susceptible to denial of service attacks, too little or the use of ballooning will hinder performance.
```
Comment (wordlee) [inline docs/administration/security/hardening-octopus.md]: Language tweaks. 
```suggestion
  - Consider the amount of memory required, if you assign too much the container is susceptible to denial of service attacks, but if you assign too little or the use of ballooning will hinder performance. 
```

File: docs/administration/security/hardening-octopus.md [hunk 4]
```diff
@@ -253,6 +253,22 @@ The TCP ports listed below are defaults, and can be changed if required - refer
...
 
+## Harden your containers
+
+If you run an [Octopus Deploy container](/docs/installation/octopus-in-container/index.md) these are a few of the steps you can take to secure it. 
+
+- Store your containers on a separate partition.
+- Assign resources carefully:
+  - Consider pinning CPUs to namespaces in order to give them a boundary. 
+  - Consider the amount of memory required, too much and the container is susceptible to denial of service attacks, too little or the use of ballooning will hinder performance. 
+- Network namespaces - consider which containers reside in each network namespace as all processes in a namespace can talk to the namespace interface.
```
Comment (wordlee) [inline docs/administration/security/hardening-octopus.md]:
```suggestion
- Consider which containers reside in each network namespace as all processes in a namespace can talk to the namespace interface.
```

File: docs/administration/security/hardening-octopus.md [hunk 5]
```diff
@@ -253,6 +253,22 @@ The TCP ports listed below are defaults, and can be changed if required - refer
...
+- Assign resources carefully:
+  - Consider pinning CPUs to namespaces in order to give them a boundary. 
+  - Consider the amount of memory required, too much and the container is susceptible to denial of service attacks, too little or the use of ballooning will hinder performance. 
+- Consider which containers reside in each network namespace as all processes in a namespace can talk to the namespace interface.
+
+The security of your containers and Docker configuration can be analyzed in detail by using [Docker Bench for Security](https://github.com/docker/docker-bench-security) from the [Center for Internet Security](https://www.cisecurity.org/about-us/). 
+
+## Samples
+
+We have an [Octopus Admin](https://g.octopushq.com/OctopusAdminSamplesSpace) Space on our Samples instance of Octopus.  Here you can sign in as `Guest` and take a look at some examples where we have used Octopus for hardening tasks.
```
Comment (wordlee) [inline docs/administration/security/hardening-octopus.md]:
```suggestion
We have an [Octopus Admin](https://g.octopushq.com/OctopusAdminSamplesSpace) Space on our Samples instance of Octopus. You can sign in as `Guest` to take a look at some examples of how we have used Octopus for hardening tasks. 
```

**Comment by wordlee** [comment]
Hi @octocrock, you need to sign the CLA (above) before we can merge the changes :)

File: docs/administration/security/hardening-octopus.md [hunk 6]
```diff
@@ -253,6 +253,22 @@ The TCP ports listed below are defaults, and can be changed if required - refer
 |WinRM-HTTP|`TCP 5985`|Octopus Server|Anywhere|DENY|Prevents attackers from using the Octopus Server as a beachhead into your network via unsecured WinRM.|
 |All outbound|`ALL`|Octopus Server|Anywhere|???|Depends on how fine-grained control you want over what your Octopus Server can do. It also depends on where your workers are, and where you are deploying to. Allowing all outbound traffic is a good place to start, then perform network analysis to decide on your next step.|
 
+## Harden your containers
+
+If you run an [Octopus Deploy container](/docs/installation/octopus-in-container/index.md), in addition to your usual security measure for running apps out of containers, take the following steps to secure it: 
+
+- Store your containers on a separate partition.
```
Comment (matt-richardson) [inline docs/administration/security/hardening-octopus.md:260]: Are you talking about the volumes? the images?
_I'm not even sure how to do this..._ (not that I'm anywhere near an expert)

File: docs/administration/security/hardening-octopus.md [hunk 7]
```diff
@@ -253,6 +253,22 @@ The TCP ports listed below are defaults, and can be changed if required - refer
...
+- Assign resources carefully:
+  - Consider pinning CPUs to namespaces in order to give them a boundary. 
+  - Consider the amount of memory required, if you assign too much the container is susceptible to denial of service attacks, but if you assign too little or make use of memory ballooning performance will be impacted. 
+- Consider which containers reside in each network namespace as all processes in a namespace can talk to the namespace interface.
+
+The security of your containers and Docker configuration can be analyzed in detail by using [Docker Bench for Security](https://github.com/docker/docker-bench-security) from the [Center for Internet Security](https://www.cisecurity.org/about-us/).
```
Comment (matt-richardson) [inline docs/administration/security/hardening-octopus.md:266]: AFAIK, this only works for linux containers.
Also, not sure how we rate on this. Might be worth checking that out first.

File: docs/installation/octopus-in-container/octopus-server-container-linux.md [hunk 8]
```diff
@@ -53,10 +53,9 @@ Read Docker [docs](https://docs.docker.com/engine/reference/commandline/run/#pub
 |  Port       |    |
 | ------------- | ------- |
 |**8080**| Port for API and HTTP portal |
+|**443**| SSL Port for API and HTTP portal |
 |**10943**|Port for Polling Tentacles to contact the server|
 
-_The Octopus Server container does not currently support HTTPS, however, this should be available sometime in the future_
```
Comment (matt-richardson) [inline docs/installation/octopus-in-container/octopus-server-container-linux.md:58]: I'm not sure we've actually made this work? Looking at the [dockerfile](https://github.com/OctopusDeploy/OctopusDeploy/blob/master/docker/linux/Dockerfile) it doesn't seem to allow it to be configured?
Users can build their own container on top of ours, but not sure we support it out of the box?

File: docs/administration/security/hardening-octopus.md (hunk 6 above)
Comment (jburger) [reply docs/administration/security/hardening-octopus.md:260]: I reckon that advice (assume it comes from here: https://www.secjuice.com/how-to-harden-docker-containers/) has got to be referring to the docker data directory that lives in `/var/lib/docker`. By moving this to a different partition, it makes it harder for a compromised container to then reach into other places in `/var` (I think)

Comment (jburger) [reply docs/administration/security/hardening-octopus.md:260]: image

Comment (jburger) [reply docs/administration/security/hardening-octopus.md:260]: How to move that folder:
https://www.guguweb.com/2019/02/07/how-to-move-docker-data-directory-to-another-location-on-ubuntu/

File: docs/administration/security/hardening-octopus.md (hunk 7 above)
Comment (jburger) [reply docs/administration/security/hardening-octopus.md:266]: I'm taking a look at this now.

Comment (jburger) [reply docs/administration/security/hardening-octopus.md:266]: Looks the purpose of this tool is to check the _Docker hosts_ security, whereas the above advice seems to indicate that it will check 'inside' your containers. @matt-richardson is correct insofar that this will only work on Linux hosts. We don't need to check ourselves against this benchmark in hosted because MS is responsible for the security of the underlying infrastructure that runs our k8s clusters.
That all said, I'd reword this slightly:
```suggestion
The security of your container host and its Docker configuration can be analyzed in detail by using [Docker Bench for Security](https://github.com/docker/docker-bench-security) from the [Center for Internet Security](https://www.cisecurity.org/about-us/). 
```

File: docs/installation/octopus-in-container/octopus-server-container-linux.md (hunk 8 above)
Comment (octocrock) [reply docs/installation/octopus-in-container/octopus-server-container-linux.md:58]: From @cvandal on Slack 

> we don’t use this any more since we moved from dedicated static ip addresses to a pool of static ip addresses, but here’s the code i wrote to support port 443, force ssl, and import the certificate: https://github.com/OctopusDeploy/OctopusDeploy/pull/4734/files

File: docs/administration/security/hardening-octopus.md (hunk 7 above)
Comment (jburger) [reply docs/administration/security/hardening-octopus.md:266]: _pretty neat tool though I ran int on my machine and it got me thinking about some stuff :)_

That tool is inspired by this organizations work, which looks like it is much less platform specific: https://www.cisecurity.org/benchmark/docker/ and could be used by a windows server admin to benchmark their own VMs/AMIs/ARMs

If you're interested in seeing the details of that benchmark I've pinned it in #school-security here https://octopusdeploy.slack.com/archives/CALSGJK7Z/p1586941721051700
